### PR TITLE
Update UI for `SitterListView` to present information about loading status and errors if needed.

### LINKED ISSUE #142: Сhange the flow view of the model to update the information on the backend
## Depends on tasks:
- #_issue

### Dev requirements and restrictions:
change the flow view of the model to update the information on the backend
- write to storage
- backend request
- after repsonce
  - if the response from the server is ok - then leave a record in the local storage
  - if the response from the server is not ok - then roll back the record in the local storage

### PR DESCRIPTION
**Pull request by NadzeyaShpakouskaya**
## Summary
This PR is a part of the work to resolve #142, #153 and #152 

## UI Preview (Screenshots)

<details>
<summary> loading indicator</summary>
<p>

</p>
</details>

<details>
<summary>error message</summary>
<p>

</p>
</details>

## Changes
- update SitterListView: add a message, if there is no available sitters.
- update SitterListView: add a loading indicator to show a user that the loading process is on.
- update SitterListView: add an error message  and `Try afain` button to attempt loading data one more time.

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/ViewModel/SitterListViewModel.swift [hunk 1]
```diff
@@ -8,17 +8,59 @@ final class SitterListViewModel: ObservableObject {
...
+                await MainActor.run {
+                    handleError(AppError.downloadFailed)
+                }
+            }
+        }
+
+        await MainActor.run { isLoading = false }
+    }
+
+    @MainActor private func handleError(_ error: Error) {
```
Comment (iBYarchitect) [inline Woof/Woof/ViewModel/SitterListViewModel.swift:57]:
```suggestion
// MARK: - Private interface

    @MainActor private func handleError(_ error: Error) {
```

**Review by iBYarchitect — APPROVED** [review]
LGTM, after solving a little mistake.

File: Woof/Woof/ViewModel/SitterListViewModel.swift [hunk 2]
```diff
@@ -8,17 +8,59 @@ final class SitterListViewModel: ObservableObject {
...
+            if let error = error as? URLError, error.code == .notConnectedToInternet {
+                await MainActor.run {
+                    handleError(AppError.noInternetConnection)
+                }
+            } else {
+                await MainActor.run {
+                    handleError(AppError.downloadFailed)
+                }
+            }
+        }
```
Comment (kirpichonok) [inline Woof/Woof/ViewModel/SitterListViewModel.swift:52]: I want to suggest a little improvement:
```suggestion
            let appError: AppError
            if (error as? URLError)?.code == .notConnectedToInternet {
                appError = .noInternetConnection
            } else {
                appError = .downloadFailed
            }
            await MainActor.run { handleError(appError) }
        }
```

File: Woof/Woof/ViewModel/SitterListViewModel.swift [hunk 3]
```diff
@@ -8,17 +8,59 @@ final class SitterListViewModel: ObservableObject {
...
+    }
+
+    @MainActor private func handleError(_ error: Error) {
+        guard let appError = error as? AppError else {
+            errorMessage = "An error occurred."
+            return
+        }
+
+        errorMessage = appError.errorDescription
+        isLoading = false
```
Comment (kirpichonok) [inline Woof/Woof/ViewModel/SitterListViewModel.swift:64]: This is redundant. We do the same in the end of the `fetchSitters`.

File: Woof/WoofTests/ViewModel/SitterListViewModelTests/SitterListViewModelTests.swift [hunk 4]
```diff
@@ -114,6 +114,54 @@ final class SitterListViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.sitters, listOfSitters)
     }
 
+    func testViewModelStateChangedToExpectedStateWhenDataFetchedSuccessfully() async throws {
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterListViewModelTests/SitterListViewModelTests.swift:117]: nit:
```suggestion
    func testViewModelChangesToExpectedStateWhenDataFetchedSuccessfully() async throws {
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -27,6 +27,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
  - Update PR template [#194](https://github.com/ios-course/ironfoudation-team-project/pull/194)
  - Update revision of the `SimpleNetworkService` package in the project - [#204](https://github.com/ios-course/ironfoudation-team-project/pull/204)
  - Update `SitterProfileView` to allow the user to enter profile info only once - [#212](https://github.com/ios-course/ironfoudation-team-project/pull/212)
+ - Update `SitterListView` to present information about loading status and errors if needed - [#218](https://github.com/ios-course/ironfoudation-team-project/pull/218)
```
Comment (kirpichonok) [inline CHANGELOG.md:30]: nit: 
```suggestion
 - Update `SitterListView` to present a loading indicator and occurring errors - [#218](https://github.com/ios-course/ironfoudation-team-project/pull/218)
```

**Review by kirpichonok — COMMENTED** [review]
Well done! 
Please, update description to PR. There is no longer `LoadingState`.
Consider a few improvements.

File: Woof/Woof/ViewModel/SitterListViewModel.swift (hunk 1 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/ViewModel/SitterListViewModel.swift:55]: Fixed

File: Woof/Woof/ViewModel/SitterListViewModel.swift (hunk 2 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/ViewModel/SitterListViewModel.swift]: Thanks for the suggestion, I fixed it with the ternary operator.

File: Woof/Woof/ViewModel/SitterListViewModel.swift (hunk 3 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/ViewModel/SitterListViewModel.swift]: Removed

File: Woof/WoofTests/ViewModel/SitterListViewModelTests/SitterListViewModelTests.swift (hunk 4 above)
Comment (NadzeyaShpakouskaya) [reply Woof/WoofTests/ViewModel/SitterListViewModelTests/SitterListViewModelTests.swift]: Fixed

File: CHANGELOG.md (hunk 5 above)
Comment (NadzeyaShpakouskaya) [reply CHANGELOG.md]: Fixed

**Review by kirpichonok — APPROVED** [review]
Great job!
LGTM

**Pull request merged**